src/store/features/board/thunks.ts

```diff
@@ -56,6 +60,50 @@ export const leaveBoard = createAsyncThunk("board/leaveBoard", async () => {
   }
 });
 
+const mapSingleParticipant = async (dto: ParticipantWithUserId): Promise<ParticipantWithUser> => {
+  try {
+    const user: Auth = await API.getUserById(dto.id);
+    return {
+      user,
+      connected: dto.connected,
+      raisedHand: dto.raisedHand,
+      ready: dto.ready,
+      showHiddenColumns: dto.showHiddenColumns,
+      role: dto.role,
+      banned: dto.banned,
+    };
+  } catch (error) {
+    // console.error("Failed to map participant with user:", error);
+    return Promise.reject(error);
+  }
+};
+
+const mapMultipleParticipants = async (dtos: ParticipantWithUserId[], boardID: string): Promise<ParticipantWithUser[]> => {
+  try {
+    const users = await API.getUsers(boardID);
+    return dtos
+      .map((dto) => {
+        const user = users.find((u: Auth) => u.id === dto.id);
+        if (!user) {
+          // console.warn(`User not found for participant ${dto.id}`);
+          return null;
+        }
+        return {user, ...dto};
+      })
+      .filter(Boolean) as ParticipantWithUser[];
+  } catch (error) {
+    // console.error("Failed to map participants with users:", error);
+    return [];
+  }
+};
+const mapParticipantsWithUsers = async (message: BoardInitEvent): Promise<ParticipantWithUser[]> => {
+  const {participants} = message.data;
+  const participantsWithId = participants as unknown as ParticipantWithUserId[];
```

Comment:
instead of this cast, you should change the property type `participants` of `BoardInitEvent` to `ParticipantWithUserId[]`.